lib/APPReinforcementLearning/src/ReadyState.cpp

```diff
@@ -59,89 +59,87 @@
 
 void ReadyState::entry()
 {
-    IDisplay&     display                 = Board::getInstance().getDisplay();
-    const int32_t SENSOR_VALUE_OUT_PERIOD = 1000; /* ms */
-
+    IDisplay& display = Board::getInstance().getDisplay();
     display.clear();
-    display.print("A: Go");
+    display.print("A: TMD");
+    display.gotoXY(0, 1);
+    display.print("B: DMD");
+
+    DifferentialDrive& diffDrive = DifferentialDrive::getInstance();
+    diffDrive.setLinearSpeed(0, 0);
 
     if (true == m_isLapTimeAvailable)
     {
-        display.gotoXY(0, 1);
+        display.gotoXY(0, 2);
         display.print(m_lapTime);
         display.print("ms");
-
-        LOG_INFO_VAL("Lap time: ", m_lapTime);
     }
-
-    /* The line sensor value shall be output on console cyclic. */
-    m_timer.start(SENSOR_VALUE_OUT_PERIOD);
+    m_modeTimeoutTimer.start(mode_selected_period);
+    m_mode                        = IDLE;
+    m_isLastStartStopLineDetected = false;
+    m_isButtonAPressed            = false;
+    m_isButtonBPressed            = false;
 }
 
 void ReadyState::process(StateMachine& sm)
 {
     IButton& buttonA = Board::getInstance().getButtonA();
+    IButton& buttonB = Board::getInstance().getButtonB();
+
+    /* Get each sensor value. */
+    ILineSensors&   lineSensors      = Board::getInstance().getLineSensors();
+    uint8_t         maxLineSensors   = lineSensors.getNumLineSensors();
+    const uint16_t* lineSensorValues = lineSensors.getSensorValues();
 
-    /* Shall track be released? */
+    /* Shall the driving mode be released? */
     if (true == Util::isButtonTriggered(buttonA, m_isButtonAPressed))
     {
-        sm.setState(&ReleaseTrackState::getInstance());
+        m_mode = DRIVING_MODE;
     }
-
-    /* Shall the line sensor values be printed out on console? */
-    if (true == m_timer.isTimeout())
+    /* Shall the Training mode be released? */
+    else if (true == Util::isButtonTriggered(buttonB, m_isButtonBPressed))
     {
-        ILineSensors&   lineSensors  = Board::getInstance().getLineSensors();
-        uint8_t         index        = 0;
-        int16_t         position     = lineSensors.readLine();
-        const uint16_t* sensorValues = lineSensors.getSensorValues();
-        char            valueStr[10];
-
-        LOG_DEBUG_HEAD();
-
-        /* Print line sensor value on console for debug purposes. */
-        for (index = 0; index < lineSensors.getNumLineSensors(); ++index)
-        {
-            if (0 < index)
-            {
-                LOG_DEBUG_MSG(" / ");
-            }
-
-            Util::uintToStr(valueStr, sizeof(valueStr), sensorValues[index]);
-
-            LOG_DEBUG_MSG(valueStr);
-        }
-
-        LOG_DEBUG_MSG(" -> ");
-
-        Util::intToStr(valueStr, sizeof(valueStr), position);
-        LOG_DEBUG_MSG(valueStr);
-
-        LOG_DEBUG_TAIL();
-
-        m_timer.restart();
+        m_mode = TRAINING_MODE;
+    }
+    else if (true == m_modeTimeoutTimer.isTimeout() && (m_mode == IDLE))
```

Comment:
Why TRAINING_MODE after timeout?

Reply:
Please add a comment to understand why this happens

Reply:
if button a or button b is not pressed it should automatically when timer is timeout, the training mode should be set and we discussed this at the previous meeting